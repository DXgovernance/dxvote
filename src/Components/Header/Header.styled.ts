import styled from 'styled-components';
import { Box, Container } from 'Components/Primitives/Layout';
import { Heading } from 'old-components/Guilds/common/Typography';

export const HeaderWrapper = styled.nav`
  padding: 0.75rem 0;

  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background: ${({ theme }) => theme.colors.background};
  z-index: 200;
  @media only screen and(min-width: 768px) {
    padding: 1.5rem 0;
  }

  border-bottom: 1px solid ${({ theme }) => theme.colors.muted};
`;

export const HeaderContainer = styled(Container)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0;
  color: ${({ theme }) => theme.colors.text};
`;

export const MenuItems = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-left: auto;
`;

export const ClickableHeading = styled(Heading)`
  cursor: pointer;
`;
