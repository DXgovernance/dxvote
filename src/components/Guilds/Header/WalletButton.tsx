import styled from 'styled-components';
import { observer } from 'mobx-react';
import { isDesktop } from 'react-device-detect';
import WalletModal from 'components/WalletModal';
import { getChains, injected } from 'provider/connectors';
import { useEffect, useState } from 'react';
import { useRpcUrls } from 'provider/providerHooks';
import { useContext } from '../../../contexts';
import { Button, IconButton } from '../common/Button';
import dxIcon from '../../../assets/images/dxdao-icon.svg';

const ButtonIcon = styled.img`
  height: 1.1rem;
  width: 1.1rem;
`;

const Web3Status = observer(() => {
  const {
    context: { modalStore, providerStore },
  } = useContext();
  const [injectedWalletAuthorized, setInjectedWalletAuthorized] =
    useState(false);
  const rpcUrls = useRpcUrls();

  useEffect(() => {
    injected.isAuthorized().then(isAuthorized => {
      setInjectedWalletAuthorized(isAuthorized);
    });
  }, [injected]);

  const { chainId, account } = providerStore.getActiveWeb3React();

  const toggleWalletModal = () => {
    modalStore.toggleWalletModal();
  };

  const switchNetwork = async chain => {
    const chainIdHex = `0x${chain.id.toString(16)}`;
    try {
      await window.ethereum?.send('wallet_switchEthereumChain', [
        { chainId: chainIdHex },
      ]);
    } catch (e: any) {
      if (e?.code == 4902) {
        window.ethereum?.send('wallet_addEthereumChain', [
          {
            chainId: chainIdHex,
            chainName: chain.displayName,
            nativeCurrency: chain.nativeAsset,
            rpcUrls: [chain.rpcUrl, chain.defaultRpc],
            blockExplorerUrls: [chain.blockExplorer],
          },
        ]);
      }
    }
  };

  function getWalletStatus() {
    if (injectedWalletAuthorized && !account) {
      const chains = getChains(rpcUrls);
      const activeChain =
        chains.find(chain => chain.id == chainId) || chains[0];
      const isMetamask = window.ethereum && window.ethereum.isMetaMask;

      return (
        <Button onClick={() => switchNetwork(activeChain)}>
          Switch {isMetamask ? 'MetaMask' : 'Wallet'} to{' '}
          {activeChain.displayName}
        </Button>
      );
    } else if (account) {
      return (
        <IconButton onClick={toggleWalletModal} iconLeft>
          <ButtonIcon src={dxIcon} alt={'Icon'} />{' '}
          {isDesktop && <span>{account}</span>}
        </IconButton>
      );
    } else {
      return <Button onClick={toggleWalletModal}>Connect Wallet</Button>;
    }
  }

  return (
    <>
      {getWalletStatus()}
      <WalletModal />
    </>
  );
});

export default Web3Status;
