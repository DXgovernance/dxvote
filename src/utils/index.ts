export * from './address';
export * from './bignumber';
export * from './cache';
export * from './constants';
export * from './date';
export * from './enums';
export * from './helperHooks';
export * from './helpers';
export * from './number';
export * from './permissions';
export * from './proposals';
export * from './scheme';
export * from './transaction';
export * from './encodingCalls';
export * from './signature';
export * from './array';
export * from './votes';
