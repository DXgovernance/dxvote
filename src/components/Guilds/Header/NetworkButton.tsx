import styled from 'styled-components';
import { observer } from 'mobx-react';
import { isChainIdSupported } from 'provider/connectors';
import NetworkModal from 'components/NetworkModal';
import { useContext } from '../../../contexts';
import { toCamelCaseString } from '../../../utils';
import { Button, IconButton } from '../common/Button';

import arbitrumIcon from '../../../assets/images/arbitrum.png';
import ethereumIcon from '../../../assets/images/ethereum.svg';
import xdaiIcon from '../../../assets/images/xdai.svg';

const iconsByChain = {
  1: ethereumIcon,
  4: ethereumIcon,
  100: xdaiIcon,
  42161: arbitrumIcon,
  421611: arbitrumIcon,
  1337: ethereumIcon,
};

const ButtonIcon = styled.img`
  height: 1.1rem;
  width: 1.1rem;
`;

const NetworkButton = observer(() => {
  const {
    context: { modalStore, providerStore, configStore },
  } = useContext();

  const { chainId, error } = providerStore.getActiveWeb3React();

  const toggleNetworkModal = () => {
    modalStore.toggleNetworkModal();
  };

  function getNetworkStatus() {
    if ((chainId && !isChainIdSupported(chainId)) || error) {
      return <Button onClick={toggleNetworkModal}>Unsupported Network</Button>;
    } else if (chainId) {
      return (
        <IconButton onClick={toggleNetworkModal} iconLeft>
          <ButtonIcon src={iconsByChain[chainId]} alt={'Icon'} />
          {toCamelCaseString(configStore.getActiveChainName())}
        </IconButton>
      );
    } else {
      return (
        <Button onClick={toggleNetworkModal} active={true}>
          Not Connected
        </Button>
      );
    }
  }

  return (
    <>
      {getNetworkStatus()}
      <NetworkModal />
    </>
  );
});

export default NetworkButton;
