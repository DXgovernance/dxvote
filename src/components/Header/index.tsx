import { withRouter } from 'react-router-dom';
import { observer } from 'mobx-react';
import styled from 'styled-components';
import Web3ConnectStatus from '../Web3ConnectStatus';
import { useContext } from '../../contexts';
import { FiSettings, FiUser, FiBarChart2 } from 'react-icons/fi';
import dxdaoIcon from 'assets/images/DXdao.svg';
import Web3 from 'web3';
import { bnum } from '../../utils';
import { Box } from '../../components/common';

const NavWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
  padding: 20px 0px 0px 0px;
`;

const NavSection = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const MenuItem = styled.div`
  display: flex;
  align-items: center;
  color: var(--nav-text-light);
  font-size: 16px;
  line-height: 19px;
  cursor: pointer;
`;

const ItemBox = styled(Box)`
  color: var(--dark-text-gray);
  padding: 5px 10px;
  font-weight: 500;
  font-size: 16px;
  margin-right: 10px;
  height: 28px;
  border-radius: 6px;
`;

const Header = observer(() => {
  const NavItem = withRouter(({ route, history, children }) => {
    return (
      <div
        style={{ cursor: 'pointer' }}
        onClick={() => {
          history.push(route);
        }}
      >
        {' '}
        {children}{' '}
      </div>
    );
  });

  const {
    context: {
      userStore,
      providerStore,
      blockchainStore,
      configStore,
      daoStore,
    },
  } = useContext();

  const { active, account } = providerStore.getActiveWeb3React();

  if (!active) {
    return (
      <NavWrapper>
        <NavSection>
          <NavItem route={`/`}>
            <MenuItem>
              <img alt="dxdao" src={dxdaoIcon} />
            </MenuItem>
          </NavItem>
        </NavSection>
        <NavSection>
          <Web3ConnectStatus text="Connect Wallet" />
          <NavItem route={`/config`}>
            <a>
              <FiSettings style={{ margin: '0px 10px', color: '#616161' }} />
            </a>
          </NavItem>
        </NavSection>
      </NavWrapper>
    );
  } else {
    const networkName = configStore.getActiveChainName();
    const userInfo = userStore.getUserInfo();
    const votingMachines = blockchainStore.initialLoadComplete
      ? configStore.getNetworkContracts().votingMachines
      : {};

    const dxdBalance =
      active && userInfo.dxdBalance
        ? parseFloat(
            Number(Web3.utils.fromWei(userInfo.dxdBalance.toString())).toFixed(
              2
            )
          )
        : 0;
    const genBalance =
      active && userInfo.genBalance
        ? parseFloat(
            Number(Web3.utils.fromWei(userInfo.genBalance.toString())).toFixed(
              2
            )
          )
        : 0;
    const { userRep, totalSupply } =
      active && blockchainStore.initialLoadComplete
        ? daoStore.getRepAt(account, providerStore.getCurrentBlockNumber())
        : { userRep: bnum(0), totalSupply: bnum(0) };
    const repPercentage = active
      ? userRep.times(100).div(totalSupply).toFixed(4)
      : bnum(0);

    return (
      <NavWrapper>
        <NavSection>
          <NavItem route={`/${networkName}/proposals`}>
            <MenuItem>
              <img alt="dxdao" src={dxdaoIcon} />
            </MenuItem>
          </NavItem>
        </NavSection>
        {blockchainStore.initialLoadComplete ? (
          <NavSection>
            {account && (
              <>
                {votingMachines.dxd ? (
                  <ItemBox> {dxdBalance} DXD </ItemBox>
                ) : (
                  <div />
                )}
                {votingMachines.gen ? (
                  <ItemBox> {genBalance} GEN </ItemBox>
                ) : (
                  <div />
                )}
                <ItemBox> {repPercentage.toString()} % REP </ItemBox>
              </>
            )}
            <Web3ConnectStatus text="Connect Wallet" />
            <NavItem route={`/${networkName}/info`}>
              <a>
                <FiBarChart2 style={{ margin: '0px 10px', color: '#616161' }} />
              </a>
            </NavItem>
            <NavItem route={`/config`}>
              <a>
                <FiSettings style={{ margin: '0px 10px', color: '#616161' }} />
              </a>
            </NavItem>
            {account && (
              <NavItem route={`/${networkName}/user/${account}`}>
                <a>
                  <FiUser style={{ margin: '0px 10px', color: '#616161' }} />
                </a>
              </NavItem>
            )}
          </NavSection>
        ) : (
          <NavSection>
            <Web3ConnectStatus text="Connect Wallet" />
            <NavItem route={`/config`}>
              <a>
                <FiSettings style={{ margin: '0px 10px', color: '#616161' }} />
              </a>
            </NavItem>
          </NavSection>
        )}
      </NavWrapper>
    );
  }
});

export default Header;
