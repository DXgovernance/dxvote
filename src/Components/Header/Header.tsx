import { useWeb3React } from '@web3-react/core';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { NetworkButton } from '../NetworkButton';
import { WalletButton } from '../WalletButton';
import {
  HeaderWrapper,
  HeaderContainer,
  MenuItems,
  ClickableHeading,
} from './Header.styled';

const Header = () => {
  const history = useHistory();
  const { active, error } = useWeb3React();
  const { t } = useTranslation();

  return (
    <HeaderWrapper as="header">
      <HeaderContainer>
        <ClickableHeading onClick={() => history.push('/')} size={2}>
          <strong>{t('guilds.guilds')}</strong>
        </ClickableHeading>
        {active && !error && (
          <MenuItems>
            <NetworkButton />
            <WalletButton />
          </MenuItems>
        )}
      </HeaderContainer>
    </HeaderWrapper>
  );
};

export default Header;
