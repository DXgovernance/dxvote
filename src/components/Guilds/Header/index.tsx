import styled from 'styled-components';
import { Box, Container } from '../common/Layout';
import { Heading } from '../common/Typography';
import NetworkButton from './NetworkButton';
import WalletButton from './WalletButton';

const HeaderWrapper = styled(Box)`
  padding: 0.75rem 0;

  @media only screen and(min-width: 768px) {
    padding: 1.5rem 0;
  }
  border-bottom: 1px solid ${({ theme }) => theme.colors.muted};
`;

const HeaderContainer = styled(Container)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0;
`;

const MenuItems = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-left: auto;
`;

const Header = () => {
  return (
    <HeaderWrapper as="header">
      <HeaderContainer>
        <Heading size={2}>
          <strong>DXvote</strong>
        </Heading>
        <MenuItems>
          <NetworkButton />
          <WalletButton />
        </MenuItems>
      </HeaderContainer>
    </HeaderWrapper>
  );
};

export default Header;
