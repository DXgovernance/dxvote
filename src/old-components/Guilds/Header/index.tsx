import { useWeb3React } from '@web3-react/core';
import { useHistory } from 'react-router-dom';
import styled from 'styled-components';
import { Box, Container } from '../../../Components/Primitives/Layout';
import { Heading } from '../common/Typography';
import NetworkButton from './NetworkButton';
import WalletButton from './WalletButton';

const HeaderWrapper = styled.nav`
  padding: 0.75rem 0;

  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background: ${({ theme }) => theme.colors.background};
  z-index: 200;
  @media only screen and(min-width: 768px) {
    padding: 1.5rem 0;
  }

  border-bottom: 1px solid ${({ theme }) => theme.colors.muted};
`;

const HeaderContainer = styled(Container)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0;
  color: ${({ theme }) => theme.colors.text};
`;

const MenuItems = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-left: auto;
`;

const ClickableHeading = styled(Heading)`
  cursor: pointer;
`;

const Header = () => {
  const history = useHistory();
  const { active, error } = useWeb3React();

  return (
    <HeaderWrapper as="header">
      <HeaderContainer>
        <ClickableHeading onClick={() => history.push('/')} size={2}>
          <strong>Guilds</strong>
        </ClickableHeading>
        {active && !error && (
          <MenuItems>
            <NetworkButton />
            <WalletButton />
          </MenuItems>
        )}
      </HeaderContainer>
    </HeaderWrapper>
  );
};

export default Header;
