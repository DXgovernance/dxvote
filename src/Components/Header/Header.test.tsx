import Header from './Header';
import { render } from 'utils/tests';

jest.mock('contexts/index', () => jest.fn());

describe('Header', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
