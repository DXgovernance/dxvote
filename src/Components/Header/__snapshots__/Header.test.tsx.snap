// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `
<div>
   
  <header
    class="sc-ehSDrC gikDCT"
  >
    <div
      class="sc-eCstlR sc-iBPTik sc-lcuiOb hzVZsK fcperG CzQAR"
    >
      <h2
        class="sc-hKgJUU sc-clsFYl edWXhv ejGdfS"
        size="2"
      >
        <strong>
          guilds.guilds
        </strong>
      </h2>
    </div>
  </header>
   
</div>
`;
